### PR TITLE
👌 Exclude external needs from `needs_id_regex` check

### PR DESCRIPTION
**Pull request by David-Le-Nir**
fix for [#1099](https://github.com/useblocks/sphinx-needs/issues/1099)

### CODE REVIEW

**Review by danwos — CHANGES_REQUESTED** [review]
Thanks for the quick PR :+1: 

The change in `need.py` looks good, as well as the updated test case.

However, I think we need to clean this PR a little bit. :)

Version changes are introduced by a separate PR. Makes it easier to rewind single commits. So please remove this.

`.DS_Store` and `sphinxcontrib-needs.iml` are also not needed.
It's okay to add them to `.gitignore` if it simplifies your life.

And it would be also good to add a small line to `docs/changelog.rst` for a `2.1.0` release.

**Comment by danwos** [comment]
Ohh and please ignore the failing pre-commit action.
Looks like a new mypy version finds much more now. I deal with this in a different PR and will merge this PR by hand.

**Comment by David-Le-Nir** [comment]
> Version changes are introduced by a separate PR. Makes it easier to rewind single commits. So please remove this.
> 
> .DS_Store and sphinxcontrib-needs.iml are also not needed.
> It's okay to add them to .gitignore if it simplifies your life.
> 
> And it would be also good to add a small line to docs/changelog.rst for a 2.1.0 release.

yes, sorry, my CLI shortcuts are a bit too "shortcutting" ... I'll fix that
yes, for the changelog, I didn't know which version to set. Also should I update the pyproject.toml to set 2.1.0 for version ?

**Comment by danwos** [comment]
No, we need to raise or better add the version just in the changelog.
All the other locations (pyproject.toml, needs.py, ...) will be done by a separate PR.

**Comment by danwos** [comment]
I was asked by @chrisjsewell to hold this PR back for some time.
He may have some concerns but is sitting on a train right now.

File: docs/changelog.rst [hunk 1]
```diff
@@ -9,6 +9,15 @@ License
 .. include:: ../LICENSE
 
 
+2.1.0
+-----
```
Comment (chrisjsewell) [inline docs/changelog.rst:13]:
```suggestion
Unreleased
----------
```

File: sphinx_needs/api/need.py [hunk 2]
```diff
@@ -190,7 +190,7 @@ def run():
     else:
         need_id = id
 
-    if needs_config.id_regex and not re.match(needs_config.id_regex, need_id):
+    if needs_config.id_regex and not is_external and not re.match(needs_config.id_regex, need_id):
         raise NeedsInvalidException(f"Given ID '{need_id}' does not match configured regex '{needs_config.id_regex}'")
```
Comment (chrisjsewell) [inline sphinx_needs/api/need.py:194]:
```suggestion
    if (
        needs_config.id_regex
        and not is_external
        and not re.match(needs_config.id_regex, need_id)
    ):
        raise NeedsInvalidException(
            f"Given ID '{need_id}' does not match configured regex '{needs_config.id_regex}'"
        )
```

**Review by chrisjsewell — CHANGES_REQUESTED** [review]
This is all good thanks, if you can apply the changes and fix the merge conflicts

File: sphinx_needs/api/need.py (hunk 2 above)
Comment (chrisjsewell) [reply sphinx_needs/api/need.py:194]: or this will be done by running pre-commit

**Review by chrisjsewell — APPROVED** [review]
cheers

**Review by danwos — APPROVED** [review]

**Pull request merged**